### ISSUE TITLE
Work with CM6?

### ISSUE THREAD

**Issue by mdbraber** [artem-barmin/obsidian-block-drag-n-drop#25]
The plugin doesn't seem to work with CM6 (on Obsidian 0.16.2). It seems the at least the package names have changes (@codemirror/gutter and @codemirror/rangeset are deprecated), but quickly changing/updating those in the base could doesn't work unfortunately. Is there a way to get this plugin to work with CM6?

**Comment by danrthompson** [comment]
Sadly +1 here. I opened an issue on the other plugin as well but it looks like this is the main plugin now.

Anyhow, happy to provide more info if that would be helpful. I'd LOVE to be able to get this working :) - awesome plugin

**Comment by artem-barmin** [comment]
Hi! Author here. Looks like 0.16 is not available in public. Newest available for download is 0.15.9. Will wait until release and will check what can be done. 

It should not be hard to fix it because I rely heavily on CM6 API. Maybe some changes in Obsidian API break things

**Comment by artem-barmin** [comment]
If you have dev experience - you can check all lines with "@ts-ignore"(only 6 of them in codebase) annotation. It's most likely reason of error and they should be adjusted to recent changes.

**Comment by danrthompson** [comment]
Ok cool I will take a look at those lines if I get a chance today. Not  familiar at all with obsidian plugins or API but if it’s an easily identifiable error I may be able to fix it or at least report what the issue is

**Comment by danrthompson** [comment]
OK so I don't know if this relates to the lines you mentioned, but the error in the console is that, as the first commenter mentioned, two packages are deprecated. I'm not 100% sure that this is the issue, as I'm not positive how to debug since nothing happens when I try to drag.

But the error messages are the following:
```
Error: [CM6][obsidian-drag-n-drop-plugin] Using a deprecated package: "@codemirror/gutter".
See the stack trace to find the faulty plugin and file an issue with the plugin author. 
Details: https://discuss.codemirror.net/t/release-0-20-0/4302
    at o (app.js:1:1034222)
    at anonymous (plugin:obsidian-drag-n-drop-plugin:5586:32)
    at e.<anonymous> (app.js:1:1034517)
    at app.js:1:235237
    at Object.next (app.js:1:235342)
    at a (app.js:1:234081)
o @ app.js:1
```
and
```
app.js:1 Error: [CM6][obsidian-drag-n-drop-plugin] Using a deprecated package: "@codemirror/rangeset".
See the stack trace to find the faulty plugin and file an issue with the plugin author. 
Details: https://discuss.codemirror.net/t/release-0-20-0/4302
    at o (app.js:1:1034222)
    at anonymous (plugin:obsidian-drag-n-drop-plugin:10928:34)
    at e.<anonymous> (app.js:1:1034517)
    at app.js:1:235237
    at Object.next (app.js:1:235342)
    at a (app.js:1:234081)
o @ app.js:1
```

I did try googling this issue and replacing the imports with the new package names, which did eliminate the error messages but unfortunately it still is not working. So perhaps I did something incorrectly, or perhaps there is a separate issue that my attempt did not resolve.

If you have any simple instructions for how to debug I'd be happy to give it a go. Also, I think the new release may have just come out (I got a new one, but I have the insider build license so I'm not 100% sure that the release is available to everyone). But whether it is out or not, I'm still more than happy to help if you'd find that useful.

**Comment by artem-barmin** [comment]
Thanks a lot for help! Obsidian 0.16 still not availble atm. 

Possible places of problem:
1. Change of gutter API for CM6. In this case we need to check examples for gutter from docs
2. Places marked with @ts-ignore - I would just check if those variables are not null. If they are CM6 instances - everything is ok

No more ideas at the moment

**Comment by micseydel** [comment]
Hi there, rusty dev here who isn't familiar with Typescript but would love to help! @danrthompson if you want, you can push your code and I can take a look.

**Comment by danrthompson** [comment]
@micseydel i dont have any code that I've written. i was just debugging a bit and trying to identify the source of the error, but I got bogged down by something else. if you can find it, that would be much appreciated 😄

**Comment by artem-barmin** [comment]
New version is out. Fix is on the way!

**Comment by artem-barmin** [comment]
Fixed in 1.2.0

**Comment by micseydel** [comment]
Thanks, I just finally got to try it out and I love it! I was confused at first by not being able to grab the bullet points, but I've just missed dragging blocks so much from Roam that I'm just so happy to have this! (Especially working on Obsidian v1.0!)
